src/transformStateWithClones.js

```diff
@@ -66,31 +66,24 @@
  */
 function transformStateWithClones(state, transforms) {
   const resultingArray = [];
+  const objTmp = { ...state };
 
-  for (let i = 0; i < transforms.length; i++) {
-    if (i === 0) {
-      resultingArray[i] = { ...state };
-    } else {
-      resultingArray[i] = { ...resultingArray[i - 1] };
-    }
-
-    for (const key in transforms[i]) {
-      if (key === 'properties') {
-        continue;
+  for (const arrayItem of transforms) {
+    if (arrayItem.operation === 'addProperties') {
+      for (const addprops in arrayItem.properties) {
+        objTmp[addprops] = arrayItem.properties[addprops];
+      }
+    } else if (arrayItem.operation === 'removeProperties') {
+      for (const propertiesItem of arrayItem.properties) {
+        delete objTmp[propertiesItem];
+      }
+    } else if (arrayItem.operation === 'clear') {
+      for (const deleteKeys in objTmp) {
+        delete objTmp[deleteKeys];
```

Comment:
deleteKeys is not a noun, so it's not a good name for a variable. Moreover, you can just write 
`objTmp = {}` instead of this loop.